src/main/java/org/commonjava/cdi/util/weft/Locker.java

```diff
@@ -19,24 +19,37 @@
 import org.slf4j.LoggerFactory;
 
 import java.util.Collections;
+import java.util.HashMap;
 import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.ReentrantLock;
 import java.util.function.BiFunction;
 import java.util.function.Function;
 
 public class Locker<K>
 {
+    private static final long DEFAULT_SWEEP_MS = 10 * 1000;
+
+    private final Timer timer = new Timer();
+
     private Map<K, ReentrantLock> locks;
 
     public Locker()
     {
-        this.locks = Collections.synchronizedMap( new ContextSensitiveWeakHashMap<>() );
+        this( Collections.synchronizedMap( new ContextSensitiveWeakHashMap<>() ), DEFAULT_SWEEP_MS );
```

Comment:
Seems synchronized twice? I think just need a "new ContextSensitiveWeakHashMap<>()" here but no need to synchronizedMap it in this cstr.